### PR TITLE
AARCH64 support

### PR DESCRIPTION
**Pull request by Avgor46**

### CODE REVIEW

**Comment by anfedotoff** [comment]
Closes #34

File: casr/tests/casr_tests/arm_bin/run.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
```
Comment (SweetVishnya) [inline casr/tests/casr_tests/arm_bin/run.sh:1]: Did you test casr-core on AARCH64?

File: casr/tests/tests.rs [hunk 2]
```diff
@@ -3063,6 +3483,176 @@ fn test_casr_afl() {
 }
 
 #[test]
+fn test_asan_stacktrace() {
```
Comment (SweetVishnya) [inline casr/tests/tests.rs:3486]: This test is in asan.rs

File: libcasr/src/gdb/exploitable.rs [hunk 3]
```diff
@@ -436,6 +441,76 @@ impl GdbContext {
...
+                return Err(Error::Casr(
+                    "Couldn't capstone instruction operands".to_string(),
+                ));
+            };
+            // Check mem operand.
+            if let capstone::arch::arm64::Arm64OperandType::Mem(_) = operand.op_type {
+                match (
+                    info.si_code,
+                    m.contains("st"),
+                    m.contains("ld"),
```
Comment (SweetVishnya) [inline libcasr/src/gdb/exploitable.rs:493]: Is this actually selects only loads and stores?

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

File: libcasr/src/gdb/exploitable.rs (hunk 3 above)
Comment (anfedotoff) [reply libcasr/src/gdb/exploitable.rs:493]: Let's use mnemonics from check_taint for load and store. Maybe we could create a functions is_load_arm64, is_store_arm64.

File: casr/tests/tests.rs (hunk 2 above)
Comment (anfedotoff) [reply casr/tests/tests.rs:3486]: Remove this test

**Review by anfedotoff — CHANGES_REQUESTED** [review]

File: libcasr/src/gdb/exploitable.rs (hunk 3 above)
Comment (Avgor46) [reply libcasr/src/gdb/exploitable.rs]: done, but mnemonics for load and store in exploitable.rs do not cover all instructions that work with memory. Should we add entire set of memory instruction?

File: casr/tests/tests.rs (hunk 2 above)
Comment (Avgor46) [reply casr/tests/tests.rs]: done

File: casr/tests/casr_tests/arm_bin/run.sh (hunk 1 above)
Comment (anfedotoff) [reply casr/tests/casr_tests/arm_bin/run.sh:1]: offline mode works fine according to tests.rs and CI

**Review by anfedotoff — APPROVED** [review]

File: libcasr/src/gdb/exploitable.rs (hunk 3 above)
Comment (anfedotoff) [reply libcasr/src/gdb/exploitable.rs]: Any examples? Only instructions that load/store registers are interested and supported by capstone also.

Comment (Avgor46) [reply libcasr/src/gdb/exploitable.rs]: STSET, CASA, LDAR?

Comment (anfedotoff) [reply libcasr/src/gdb/exploitable.rs]: STSET - is bit set (suppose no), CAS -compare and swap (suppose no), LDAR - maybe, If you think it's necessary you could and it.

**Review by SweetVishnya — APPROVED** [review]

**Pull request merged**